tests_integration/test_connector_flows.py

```diff
@@ -208,39 +184,31 @@ def test_connector_flows_pod_gone_while_operator_down(run_kopf, random_connector
 
     with run_kopf():
         kubectl_create(OBJ)
-        time.sleep(10)
-
+        kubectl_wait_to_exist("secret", connector_name)
+        kubectl_wait_pod_running(connector_name)
         connector = kubectl_get("tc", connector_name)
-        kubectl_get("secret", connector_name)
-        pod = kubectl_get("pod", connector_name)
-
-        while pod["status"]["phase"] == "Pending":
-            time.sleep(1)
-            pod = kubectl_get("pod", connector_name)
 
         # connector was properly provisioned
         expected_status = {"success": True, "ts": ANY, "twingate_id": ANY}
         assert connector["status"]["twingate_connector_create"] == expected_status
 
-    kubectl_delete(f"pod/{connector_name}")
+    kubectl_delete("pod", connector_name)
 
     # run operator again
     with run_kopf():
-        time.sleep(5)
-
         # pod was recreated
-        pod = kubectl_get("pod", connector_name)
+        pod = kubectl_wait_pod_running(connector_name)
         assert pod["status"]["phase"] == "Running"
 
         # Test done, delete connector
-        kubectl_delete(f"tc/{connector_name}")
+        kubectl_delete("tc", connector_name)
         time.sleep(5)
```

Comment:
Should we also use `kubectl_delete_wait` here?

Reply:
no because after `tc` deletion is done we still need k8s to delete the secret and pod and `kubectl_delete_wait` wont do that